src/main/java/QueryServlet.java

```diff
@@ -43,8 +47,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             "https://api.census.gov/data/2018/acs/acs1/spp?get=NAME,S0201_"
                 + queryToDataRow.get(action).get(personType)
                 + "&for="
-                + location
-                + ":*&key=ea65020114ffc1e71e760341a0285f99e73eabbc");
+                + locationTypeToQuery.get(location)
+                + stateNumber
```

Comment:
It looks like when the query is for all states, you set the state-number parameter to an empty string in the JS so that it can always be appended here. This definitely works, but I wonder if it would make more sense from an error handling/avoidance perspective to do it a bit differently? For example, not including the state-number parameter if we didn't ask for states, or making it equal to false or something, and then only appending the stateNumber string if there was an actual number.

Reply:
Yeah, I agree the way I had it before didn't make much sense. For now I have the values for CA and NJ hardcoded in, will have to figure out how to do this dynamically for each state